commands/create.go

```diff
@@ -458,3 +483,95 @@ func tlsPath(c CommandLine, flag string, defaultName string) string {
 
 	return filepath.Join(mcndirs.GetMachineCertDir(), defaultName)
 }
+
+func updateUserdataFile(driverOpts *rpcdriver.RPCFlags, userdataFlag, customInstallScript string) error {
+	var userdataContent []byte
+	var err error
+	userdataFile := driverOpts.String(userdataFlag)
+
+	if userdataFile == "" {
+		if strings.HasPrefix(userdataFlag, "cloud-config") {
+			// Drivers like vmwarevsphere expect the userdata to be in cloud-config format.
+			// Setting userdataContent like this will coerce the custom script into cloud-config format.
+			userdataContent = []byte("#cloud-config")
+		} else {
+			// If the userdata file is not set, then the customInstallScript file is set as the userdata file.
+			driverOpts.Values[userdataFlag] = customInstallScript
+			return nil
+		}
+	} else {
+		userdataContent, err = ioutil.ReadFile(userdataFile)
+		if err != nil {
+			return err
+		}
+	}
+
+	customScriptContent, err := ioutil.ReadFile(customInstallScript)
+	if err != nil {
+		return err
+	}
+	customScriptContent = regexp.MustCompile(`^#!.*\n`).ReplaceAll(customScriptContent, nil)
+
+	// TODO: maybe not make this a temp file?
+	modifiedUserdataFile, err := ioutil.TempFile("", "modified-user-data")
+	if err != nil {
+		return err
+	}
+	defer modifiedUserdataFile.Close()
+
+	err = replaceUserdataFile(userdataContent, customScriptContent, modifiedUserdataFile)
+	if err == nil {
+		driverOpts.Values[userdataFlag] = modifiedUserdataFile.Name()
+	}
+
+	return err
+}
+
+func replaceUserdataFile(userdataContent, customScriptContent []byte, newUserDataFile *os.File) error {
+	var err error
```

Comment:
nit: Probably best to just remove this, return the errors where they occur and return nil at the bottom. In my opinion it's good to just return when feasible, it's easy to read and trace what's going on.

Reply:
Do you mean the same as the comment below?

Reply:
Will do.